File: server/datasets/python_source-code/Multiple-A1/organic6_binary_search.py
"""
Binary Search - Submission by Patricia
Approach: Delegates the heavy lifting to Python's built-in bisect
module (bisect_left) instead of hand-rolling the loop.
"""

import bisect


def find_index(arr, target):
    pos = bisect.bisect_left(arr, target)
    if pos < len(arr) and arr[pos] == target:
        return pos
    return -1


if __name__ == "__main__":
    ARR = [2, 5, 8, 12, 16, 23, 38, 45, 56, 72, 91]
    TARGET = 23
    print("bisect result ->", find_index(ARR, TARGET))

File: server/datasets/controlled_benchmarks/java/multi_student_cohorts/cohort_a1/organic6_binary_search.java
/*
 * Binary Search - Submission by Patricia
 * Approach: Delegates search to standard library (Arrays.binarySearch)
 * instead of hand-rolling the loop.
 */

import java.util.Arrays;

class BinarySearch {
    public static int findIndex(int[] arr, int target) {
        int pos = Arrays.binarySearch(arr, target);
        if (pos >= 0 && pos < arr.length && arr[pos] == target) {
            return pos;
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] arr = {2, 5, 8, 12, 16, 23, 38, 45, 56, 72, 91};
        int target = 23;
        System.out.println("bisect result -> " + findIndex(arr, target));
    }
}
